### ISSUE TITLE
Fails to encrypt/decrypt second time

### ISSUE THREAD

**Issue by dfahlander** [mark43/dexie-encrypted#5]
My application code looks like this:

FriendApi.ts

```ts
import { Friend } from "./FriendList";
import Dexie from 'dexie';
import encrypt, { cryptoOptions } from 'dexie-encrypted';

const db = new Dexie("friendsDB4");

const binary_string = atob("pDA+nXDxXychsH/DQlBd/1b+4y0cjd1dsqRQwKiDD3U=");
const len = binary_string.length;
const cryptoKey = new Uint8Array(len);
for (var i = 0; i < len; i++) {
  cryptoKey[i] = binary_string.charCodeAt(i);
}

encrypt(db, cryptoKey, {friends: cryptoOptions.NON_INDEXED_FIELDS});

db.version(1).stores({
  friends: 'id, name'
});

export class FriendApi {
  friends = db.table("friends") as Dexie.Table<Friend, number>;

  listFriends (): Promise<Friend[]> {
    return this.friends.orderBy("name").toArray();
  }

  upsertFriend(friend: Friend) {
    return this.friends.put(friend);
  }

  removeFriend(id: number) {
    return this.friends.delete(id);
  }
}

export const friendApi = new FriendApi();

```

Now, if I add a friend, it works perfect, but if I edit the friend, it will alway get `age: 0`. The field "age" is the only non-indexed field here.

**Comment by stutrek** [comment]
I believe this issue should be fixed by #3 in 1.0.2. It was because the object going into the database must be mutated then mutated back to its original form.

**Comment by lifter035** [comment]
I'm having the same issue with 1.0.3. When adding new record everything works perfectly but when using .put or .bulkPut to update the record it seems that encrypted data is all set to default empty strings, 0 numbers and false booleans

**Comment by stutrek** [comment]
When you fetch it from the database it will be populated with real data. Dexie hooks use mutation so I chose to zero everything out instead of deleting to allow the browser to optimize more efficiently.

**Comment by lifter035** [comment]
After it is put in I do fetch everything again and data received is still zeroed out. Even if I refresh the page which loads everything again

**Comment by stutrek** [comment]
I managed to reproduce this, trying to write a test that gets it consistently right now.

**Comment by lifter035** [comment]
Ah cool I added stackblitz that reproduces:
https://stackblitz.com/edit/typescript-kumued?file=index.ts

If reloading delete the db

**Comment by stutrek** [comment]
Just published 1.0.4 with a fix!

The update hook needs to decrypt then encrypt the updates.

**Comment by lifter035** [comment]
Awesome thanks 👍
